parallaxinc/solo
# Refactor blocks using hidden fields

Addresses #113 
Se also: [google/blockly/issues/3223](https://github.com/google/blockly/issues/3223)

**This affects the following blocks:**
comment
string_split
wx_init_simple (experimental)
wx_init_advanced
wx_scan_multiple
wx_scan_string
wx_connect
wx_mode
adc_read (FLiP or Other type only)
gps_read

custom_code is also affected, but it needs more extensive work to update it's fields - changes to the field API in the blockly core have and will render this block very broken, and it probably needs its own issue.

**To Test:**
1. drag the affected block to the workspace
2. modify the fields until you find one that changes the shape of the block by hiding/showing fields
3. perform other actions on the block like dragging, editing a field, etc. after the block shape has changed.
a. Make sure field values are not strangely modified (in wx_connect, a variable name is changed intentionally)
b. Open the console to make sure dragging/interacting with the block does not generate errors or warnings
4. Save, close, re-open to make sure the block state/values persist
